skypy/galaxy/tests/test_spectrum.py

```diff
@@ -51,3 +53,34 @@ def test_sampling_coefficients():
     with pytest.raises(ValueError,
                        match="alpha0 and alpha1 must be array_like."):
         dirichlet_coefficients(redshift, alpha0, scalar_alpha)
+
+
+def test_kcorrect_spectra():
+    # Test that the shape of the returned flux density corresponds to (nz, nl)
+    kcorrect_templates_url = "https://github.com/blanton144/kcorrect/raw/" \
+                             "master/data/templates/k_nmf_derived.default.fits"
+    lam = getdata(kcorrect_templates_url, 11)
+    coefficients = np.array([[1, 1, 1, 1, 1],
+                            [1, 1, 1, 1, 1]])
+    z = np.array([0.5, 1])
+    mass = np.array([5 * 10 ** 10, 7 * 10 ** 9])
+    lam_o, sed = kcorrect_spectra(z, mass, coefficients)
+
+    assert sed.shape == (len(z), len(lam))
+
+    # Test it returns the right sed and wavelength
+    templates = getdata(kcorrect_templates_url, 1)
+
+    coefficients = np.array([1, 0, 0, 0, 0])
+
+    z = np.array([0])
+    mass = np.array([1])
+
+    sed_test = np.array([templates[0], templates[0]])
+    lam_o_test = np.matmul((1 + z).reshape(len(z), 1),
+                           lam.reshape(1, len(lam)))
+
+    lam_o, sed = kcorrect_spectra(z, mass, coefficients)
+
+    assert np.allclose(lam_o, lam_o_test)
+    assert np.allclose(sed, sed_test)
```

Comment:
```suggestion
def test_kcorrect_spectra():
    # Download template data
    kcorrect_templates_url = "https://github.com/blanton144/kcorrect/raw/" \
                             "master/data/templates/k_nmf_derived.default.fits"
    lam = getdata(kcorrect_templates_url, 11)
    templates = getdata(kcorrect_templates_url, 1)
    
    # Test that the shape of the returned flux density corresponds to (nz, nl)
    coefficients = np.array([[0.2, 0.2, 0.2, 0.2, 0.2],
                            [0.0, 0.1, 0.2, 0.3, 0.4]])
    z = np.array([0.5, 1])
    mass = np.array([5e10, 7e9])
    lam_o, sed = kcorrect_spectra(z, mass, coefficients)
    assert sed.shape == (len(z), len(lam))

    # Test that for redshift=0, mass=1 and coefficients=[1,0,0,0,0]
    # the returned wavelengths and spectra match the template data

    coefficients = np.array([1, 0, 0, 0, 0])
    z = np.array([0])
    mass = np.array([1])
    lam_o, sed = kcorrect_spectra(z, mass, coefficients)

    assert np.allclose(lam_o, lam)
    assert np.allclose(sed, templates[0])
```
- Reorder the code and clearly explain each test
- Use "valid" coefficients that sum to one
- No need to recalculate the wavelengths at redshift zero
- `sed_test` was wrong just compare against templates[0]